### PR TITLE
test-e2e as a composite action

### PR DESCRIPTION
**Pull request by mundra-ankur**
Signed-off-by: Ankur Mundra <amundra@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 GOBIN ?= ${GOPATH}/bin
 IMG   ?= tackle2-addon-windup:latest
-CONTAINER_RUNTIME := $(shell command -v podman 2> /dev/null || echo docker)
+CONTAINER_RUNTIME := docker
```
Comment (djzager) [inline Makefile:3]: Same as https://github.com/konveyor/tackle2-hub/pull/177#discussion_r1000898275 either we "discover" it or drop it entirely.

File: Makefile [hunk 2]
```diff
@@ -40,9 +40,14 @@ ifeq (,$(wildcard $(INSTALL_TACKLE_SH)))
...
-	export export TACKLE_ADDON_WINDUP_IMAGE=$(IMG); $(INSTALL_TACKLE_SH);
+	export TACKLE_ADDON_WINDUP_IMAGE=$(IMG); $(INSTALL_TACKLE_SH);
 
 .PHONY: test-e2e
 test-e2e: start-minikube build-image install-tackle; \
 	export HOST=http://$(shell minikube ip)/hub; \
 	bash hack/test-e2e.sh;
+
+.PHONY: test-e2e-without-env-setup
+test-e2e-without-env-setup:
```
Comment (djzager) [inline Makefile:51]: Is it possible to, only have `test-e2e` (without the dependency on `start-minikube` + `build-image` + `install-tackle`). Then update https://github.com/konveyor/tackle2-addon-windup/blob/22ef356381a2fd44fc48f89d1f60f5b525007f30/.github/workflows/test-windup.yml#L35-L40 to something like:

```
      - name: start-minikube
        uses: konveyor/tackle2-operator/.github/actions/start-minikube@main
      - name: Build image in minikube
        run: |
          export SHELL=/bin/bash
          eval $(minikube -p minikube docker-env)
          make build-image
      - name: install-tackle
        uses: konveyor/tackle2-operator/.github/actions/install-tackle@main
```

File: Makefile (hunk 1 above)
Comment (mundra-ankur) [reply Makefile]: Added separate target for docker and podman

File: Makefile (hunk 2 above)
Comment (mundra-ankur) [reply Makefile]: Done!

**Review by djzager — APPROVED** [review]

**Pull request merged**